homeassistant/components/nx584/alarm_control_panel.py

```diff
@@ -45,12 +58,35 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     url = f"http://{host}:{port}"
 
+    entity = NX584Alarm(hass, url, name)
+
     try:
-        add_entities([NX584Alarm(hass, url, name)])
+        add_entities([entity])
     except requests.exceptions.ConnectionError as ex:
         _LOGGER.error("Unable to connect to NX584: %s", str(ex))
         return
 
+    def alarm_bypass_handler(service):
+        """Register zone bypass handler."""
+        zone = service.data.get(ATTR_ZONE)
+        entity.alarm_bypass(zone)
+
+    def alarm_unbypass_handler(service):
+        """Register zone unbypass handler."""
+        zone = service.data.get(ATTR_ZONE)
+        entity.alarm_unbypass(zone)
+
+    hass.services.register(
```

Comment:
Please use our entity service helper instead:
https://developers.home-assistant.io/docs/dev_101_services#entity-services

Reply:
Will do.  Wish I knew about that before.  Sorry, just diving into HA development.

Reply:
Okay, switched it to using entity service helper.  Let me know if you want anything else changed.  Thanks!